nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java

```diff
@@ -80,12 +79,14 @@
   private UpdateableReference reference;
   private String name;
   private FileIO fileIO;
+  private Map<String, String> catalogOptions;
 
   public NessieCatalog() {
   }
 
   @Override
   public void initialize(String inputName, Map<String, String> options) {
+    this.catalogOptions = ImmutableMap.copyOf(options);
```

Comment:
Does this map actually have the `spark` config options? I thought it was just the subset of catalog options?

Reply:
I believe that only a subset of that config is being passed down. Only the `spark.sql.catalog.catalog_name.*` options get passed through.

Reply:
I have an integration test with Spark in https://github.com/apache/iceberg/commit/dac4ab8b2dcb4faf4e974a0398dad0d4a4c84c82#diff-a69dddda28c4edb1f19f2f9e84de0e74a76fcf9e231b6975a72fab193b003733R146 that makes sure that the `spark.app.id` is set correctly. Unfortunately I can't commit those changes atm, because it  requires a few other changes which are also not yet committed to `master`. Given that, it seems those options are passed down properly to the `NessieCatalog`, unless I'm missing some logic where those things are explicitly filtered out and not passed further down?

Reply:
I agree with @rymurr. The properties passed as `options` in the Spark catalog API are the ones under the catalog name and not general options (unless Spark added this and I don't know about it).

I think it would be better to ensure that the application ID is taken from the Spark context in the [`SparkCatalog` implementation](https://github.com/apache/iceberg/blob/master/spark3/src/main/java/org/apache/iceberg/spark/SparkCatalog.java#L99). That has already called `SparkSession.active()` so you can get the ID from it directly and add it as an option to the property map. By doing it that way and using a common name, like `app-id`, to pass the app ID, we can set a convention that could also be used by Flink to set the app ID context.

Reply:
Nevermind, it looks like this forwarding was already done in SparkCatalog below. I think that it's cleaner to do it in `buildIcebergCatalog`, though. That way you don't have to re-create the `CaseInsensitiveStringMap`.

Reply:
sgtm, will move stuff to `buildIcebergCatalog`